docs/merged_reduced_scans_table.md

```diff
@@ -0,0 +1,122 @@
+| read        | Reading from the remote |
+| http        | Verification of HTTP headers. `HTTP/S` only |
+| content     | Verification that the returned content matches the expected content. These are the most common types of errors and represent things like blockpages. |
+| final       | Reaching the final stage without encountering any problems in the previous stages. |
+| unknown     | The stage is not known. Usually these are new errors which should be investigated and classified  |
+
+#### Stages per Probe
+
+Not all tests include every stage depending on the type of test. For example since the Echo protocol does not involve TLS Echo tests will never fail with errors classified as `tls`. Here are the stages for each test type.
+
+##### Discard
+
+![discard connection diagram](diagrams/discard.svg)
+
+##### Echo
+
+![echo connection diagram](diagrams/echo.svg)
+
+##### HTTP
+
+![http connection diagram](diagrams/http.svg)
+
+##### HTTPS
+
+![https connection diagram](diagrams/https.svg)
+
+### Error Classes
+
+Basic errors represent simplest types of errors, as well as the `success` case (no error detected).
+
+Protocol errors are similar but not identical to the Network Error Logging standard's [Predefined Network Error Types](https://www.w3.org/TR/network-error-logging/#predefined-network-error-types). These errors may represent normal network failures and noise, but they may also expose interference by a middlebox to disrupt a connection. For example China and Iran are both known to use [TCP Reset Attacks](https://en.wikipedia.org/wiki/TCP_reset_attack) as a censorship method.
+
+Mismatch Errors are used when the connection is successful, but the content received does not match the content expected. This can happen in the case of blockpages, remote servers with unusual behavior, or complicated CDN networks serving many sites.
+
+| Error Class             | Explanation |
+| ----------------------- | ----------- |
+|                         |
+| **Basic Errors**        |
+|                         |
+| success                 | The test completed successfully and no interference was detected |
+| system                  | There was a test system failure, rendering the test invalid |
+| unknown                 | The class of the error is not known. Usually these are new errors which should be investigated and classified |
+|                         |
+| **Protocol Errors**     | There were errors in the connection protocol |
+|                         |
+| ip.address_unreachable  | The IP address is unreachable |
+| ip.no_route             | No route to the IP address could be found |
+| timeout                 | The connection timed out. Could indicate packets being dropped by a middlebox |
+| tcp.refused             | The TCP connection was refused by the server |
+| tcp.closed              | The TCP connection was closed by the server. Could indicate an inserted `FIN` packet |
```

Comment:
I'm not still not comfortable with this because it's not a TCP error. Closing a TCP connection is fine and expected.

The problem is closing the connection when not expected. From the screenshot you sent, it's coming from the the HTTP fetching. EOF means error == EOF and n == 0, an empty response. unexpected EOF is when n > 0 when the HTTP parsing was expecting more data.

I'd say lets map EOF -> http.empty and unexpectedEOF -> http.unexpected_eof or http.truncated_response.

Both cases are actually instances of malformed HTTP, that's why I believe it belongs under http, not tcp.